TwentyTwentyOne: Add bottom-border to hover states in pagination links

allancole opened the pull request:
<!-- Add reference to the issue this pull-request fixes.-->
Fixes: #868

## Summary

This PR simply adds bottom-borders to the hover state on pagination links to make them match the current page bottom-border. 

## Test instructions

This PR can be tested by following these steps:
1. Visit any post archive page (search, category, etc.)
2. Scroll to the bottom and hover over the pagination links.
3. You should see a thin underline beneath the hovered link that matches the bottom-border for the _current_ page text. (see screenshots below)
<!-- Don't forget to test the unhappy-paths! -->

## Screenshots
Before: 
![2020-11-23 16 32 03](https://user-images.githubusercontent.com/709581/100017785-71637b00-2da9-11eb-9fce-823e15dbdf00.gif)

After: 
![2020-11-23 16 30 25](https://user-images.githubusercontent.com/709581/100017667-4416cd00-2da9-11eb-891e-1c5cf35f1d78.gif)

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

allancole committed:
TwentyTwentyOne: Add bottom borders to hover states in paginantion links.

melchoyce commented:
LGTM 👍

carolinan commented:
I don't think these borders would match any other link hover styles? I don't think it looks bad but it will be inconsistent?

melchoyce commented:
Oh, you're right — I had them reversed in my head 😵 💫  Ideally the post navigation styles would match the menu styles: solid action, dotted hover:

![image](https://user-images.githubusercontent.com/2846578/100030548-c57a5980-2dc1-11eb-8831-0f32bc94d2f2.png)

allancole commented:
Ahhh, I didn’t catch that! I’ll reverse this so it matches the menu styles.

allancole committed:
TwentyTwentyOne: Make pagintaion hover/active underlines match the primary navigation.

allancole commented:
Should be good to go now @melchoyce & @carolinan: 

Header Nav:
![2020-11-24 11 28 32](https://user-images.githubusercontent.com/709581/100123228-4714ca80-2e48-11eb-8410-221fe2ddb9a5.gif)


Pagination: 
![2020-11-24 11 27 14](https://user-images.githubusercontent.com/709581/100122995-0d43c400-2e48-11eb-9a34-667cf3092df8.gif)

allancole commented:
Also just noticed the font-sizes don’t match between the header navigation and pagination links:

20px header nav| 24px pagination
------------|------------
![image](https://user-images.githubusercontent.com/709581/100123429-7deae080-2e48-11eb-8d19-9d5cd8d09942.png)|![image](https://user-images.githubusercontent.com/709581/100123483-8e02c000-2e48-11eb-9d0c-8be6cc848e59.png)

Since the hover actions are the same, should the font-sizes also the same @melchoyce ?

melchoyce commented:
I think I want to leave them as-is. Thanks for checking!

ryelle commented:
I noticed that the current item has the hover state too, but it's not a link. That could be confusing, I'm not sure if it's technically an a11y issue or not. Could `.current` also be excluded from the hover state?

carolinan commented on assets/sass/06-components/pagination.scss:
```diff
@@ -164,7 +164,11 @@
+			text-decoration: underline;
+		}
+
+		&:not(.dots):hover {
```
```suggestion
		&:not(.dots):not(.current):hover {
```

carolinan committed:
Update assets/sass/06-components/pagination.scss

carolinan commented:
Just a note that with this change we have solid underlines for next and previous posts (single view), but dotted underlines for the newer/older post pagination. @melchoyce Is that what you want?

allancole commented:
> Could .current also be excluded from the hover state?

Ahh, good call. I’ll correct that :-)

allancole committed:
TwentyTwentyOne: Fix underline styles in next/prev links
- Adds dotted underline to singlar next/prev links
- Removes hover state from current page number since it isn’t a link

allancole committed:
TwentyTwentyOne: Recompile.

allancole committed:
Merge branch 'fix/issue-868' of github.com:WordPress/twentytwentyone into fix/issue-868

allancole commented:
> Just a note that with this change we have solid underlines for next and previous posts (single view), but dotted underlines for the newer/older post pagination. @melchoyce Is that what you want?

I went ahead and corrected this in that last batch of commits. Feels more consistent this way but happy to change it back if you prefer, @melchoyce. Cc: @carolinan

melchoyce commented:
Can you add another gif for comparison @allancole?

allancole commented:
No prob, @melchoyce. See below:

### Archive View
Before: 
![2020-11-30 16 24 40](https://user-images.githubusercontent.com/709581/100667485-91a2b500-3328-11eb-8348-7aaa243a8096.gif)

After:
![2020-11-30 16 25 25](https://user-images.githubusercontent.com/709581/100667589-ab43fc80-3328-11eb-86c9-93e399868e84.gif)

### Singular View
Before: 
![2020-11-30 16 23 30](https://user-images.githubusercontent.com/709581/100667353-67e98e00-3328-11eb-9167-476788dd8ece.gif)

After:
![2020-11-30 16 18 28](https://user-images.githubusercontent.com/709581/100666876-c2361f00-3327-11eb-917f-eadb8444f48f.gif)

pattonwebz commented:
The _after_ gifs look real nice to me :heart:

melchoyce commented:
Same, thanks @allancole 👍

allancole commented:
I think this is good to merge then, @carolinan—if you’d like to do the honors ;-)

carolinan commented:
I am very confused about why one of the after gifs have both a dotted and solid underline?
